src/lock.ts

```diff
@@ -1,16 +1,15 @@
 import Connector from './connector';
 
 export default class Lock {
-  public connectors = {};
+  public connectors: Record<string, Connector> = {};
   public options = {};
 
   addConnector(connector: any) {
-    this.connectors[connector.key] = connector.connector;
+    this.connectors[connector.key] = new connector.connector(connector.options);
     this.options[connector.key] = connector.options;
```

Comment:
With these changes, we won't need `this.options` right?

Reply:
Maybe, but it's a public property, so not sure if some implementation are accessing it directly